### PR TITLE
[CSL 360] Add support for Facet Option configurations

### PR DESCRIPTION
**Pull request by jjl014**
Updates:
* Add support for Facet Option Configurations
* Add tests, cleans up after itself (all pass)

### CODE REVIEW

File: spec/src/modules/catalog.js [hunk 1]
```diff
@@ -3120,4 +3135,453 @@ describe('ConstructorIO - Catalog', () => {
...
+      it('Should resolve when adding facet configurations', (done) => {
+        const { catalog } = new ConstructorIO({
+          ...validOptions,
+          fetch: fetchSpy,
+        });
+
+        catalog.addOrModifyFacetOptionConfigurations({
+          facetGroupName,
+          facetOptionConfigurations: mockFacetOptionConfigurations,
+        }).then(() => done());
```
Comment (esezen) [inline spec/src/modules/catalog.js:3223]: I think this `.then(() => done());` can be `.then(done);`

**Review by esezen — APPROVED** [review]
The tests are awesome! Great work 🚀

File: spec/src/modules/catalog.js (hunk 1 above)
Comment (jjl014) [reply spec/src/modules/catalog.js:3223]: Jus adding a comment here for others. `done` isn't expecting any arguments to be passed to it and errors if there is. So we have to leave it has `.then(() => done())`.

**Comment by sblaurock** [comment]
@jjl014 a heads up that I updated some of the JSDoc descriptions to make use of specifying defaults directly in the documentation rather than in the description: https://jsdoc.app/tags-param.html#optional-parameters-and-default-values

This PR looks fantastic - great work!!!

**Pull request merged**